### DISCUSSION TITLE
Cannot add Sonoff ZBMINIL2 to groups

### DISCUSSION THREAD

**Discussion by ukmgranger** [Koenkk/zigbee2mqtt#22705] [Devices]
Hi all - I have two ZBMINIL2 devices in my setup, but it seems that I cannot add either of them to any of my groups.  I get the following error:
`Failed to add from group (ZCL command 0x70ac08fffebfdc05/1 genGroups.add({"groupid":2,"groupname":""}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Timeout - 65107 - 1 - 50 - 4 - 0 after 10000ms))`

As a note, the devices both function well, and I have tried activating the devices just before attempting to add them to a group.

Any ideas?

**Comment by marcbernis** [comment]
Hi have same on my 6 ZBMINIL2 devices.
ZBMINIL2 clusters do not support genGroups.
I do not know if it is because Sonoff firmware does not support it (probably) or Z2M implementation.
So nothing to do on this as fas as I know.

**Reply by ukmgranger** [reply]
Any further advice on this?  I forgot that I can't add these to groups and went a bought another not thinking :(

**Reply by marcbernis** [reply]
My understanding is that this is a ZBMINIL2 firmware limitation and nothing can be done on Z2M side.
It is unfortunate as it usually command lights  and having a group should be a basic.

**Comment by SilentImp** [comment]
Experiencing same issue

**Comment by sschuepfer** [comment]
same problem

**Comment by samsam-rolon** [comment]
Same issue there

**Comment by andrei-lazarov** [comment]
I believe it's a **Zigbee** limitation. For some reason you can't use **end_devices** in groups.
Tried with multiple devices (relays). Even if adding to the group works, it won't respond to any group commands.

**Comment by leonopit2** [comment]
Same issue, same frustration with the ZBMINIL2.

**Comment by marcbernis** [comment]
Hello,
Checked sonoff tech page
https://help.sonoff.tech/docs/zbminil2
In key features middle page they state :

Create Group

After multiple devices are created into a group, you can synchronize all
devices in the group.

But the group cluster is not available in Z2M.

**Comment by samsam-rolon** [comment]
On Dec,24th 2025, SonOff Responds me this :
Dear Sam,

Thanks for your email.
To avoid unexpected shutdowns due to insufficient power, single-fire devices need to be as energy-efficient as possible and will enter sleep mode; thus, they do not support group operation.

Thanks for your time! 

Regards,
Aron in SONOFF
